src/devices_models/devices/hybrid_generation.jl

```diff
@@ -167,6 +167,113 @@ get_min_max_limits(
     ::Type{<:AbstractHybridFormulation},
 ) = PSY.get_state_of_charge_limits(PSY.get_storage(device))
 
+# TODO: Refactor these constraints to be used only by HybridSystems methods and change name
+# of the function
+
+function _add_lower_bound_range_constraints_impl!(
+    container::OptimizationContainer,
+    T::Type{ComponentActivePowerVariableLimitsConstraint},
+    array,
+    devices::IS.FlattenIteratorWrapper{V},
+    model::DeviceModel{V, W},
+) where {V <: PSY.HybridSystem, W <: AbstractDeviceFormulation}
+    constraint = T()
+    component_type = V
+    time_steps = get_time_steps(container)
+    device_names =
+        [PSY.get_name(d) for d in devices if does_subcomponent_exist(d, PSY.ThermalGen)]
```

Comment:
as a rule of thumb if you need to call `does_subcomponent_exist` that function needs to be called here